### PR TITLE
ENH: Filter will make output directory if needed.

### PR DESCRIPTION
**Pull request by imikejackson**
ENH: Using unique error codes which make it easier for the developer to
track errors.

### CODE REVIEW

**Comment by fbudin69500** [comment]
Couldn't the user use the browsing window to create a directory if needed? Maybe my comment is too late since you already wrote the patch :)

**Comment by imikejackson** [comment]
But if the pipeline is being run from the command line and the user has put in an output directory that  does not exist my assumption is that they want this path made. In this use case there is not GUI for them to create the directory structure. We do this in our other filters so it has precedent for being done in DREAM.3D.

**Comment by imikejackson** [comment]
Apparently some code formatting got in there also. I can see if I can pull that out of the patch if you want me to.

**Comment by fbudin69500** [comment]
@imikejackson This makes perfect sense. I did not anticipate the command line usage.

File: ITKImageProcessingFilters/ITKImageWriter.cpp [hunk 1]
```diff
@@ -127,10 +129,19 @@ void ITKImageWriter::dataCheck()
     return;
   }
 
+  QFileInfo fi(filename);
+  QDir parentPath(fi.path());
+  if(parentPath.exists() == false)
+  {
+    setErrorCondition(-21001);
```
Comment (fbudin69500) [inline ITKImageProcessingFilters/ITKImageWriter.cpp]: is 'setErrorCondition' used also for warnings?

**Comment by thewtex** [comment]
+1

**Review by fbudin69500 — APPROVED** [review]

File: ITKImageProcessingFilters/ITKImageWriter.cpp (hunk 1 above)
Comment (imikejackson) [reply ITKImageProcessingFilters/ITKImageWriter.cpp]: No it is not. I'll push another commit to fix that. Good find. Thanks

**Pull request merged**